src/proxy/cpsubsystem/BaseCPProxy.ts

```diff
@@ -15,35 +15,41 @@
  */
 /** @ignore *//** */
 
-import {HazelcastClient} from '../../HazelcastClient';
 import {ClientMessage} from '../../protocol/ClientMessage';
 import {RaftGroupId} from './RaftGroupId';
 import {CPGroupDestroyCPObjectCodec} from '../../codec/CPGroupDestroyCPObjectCodec';
 import {UnsupportedOperationError} from '../../core';
 import {Data} from '../../serialization/Data';
+import {SerializationService} from '../../serialization/SerializationService';
+import {InvocationService} from '../../invocation/InvocationService';
 
 /**
  * Common super class for any CP Subsystem proxy.
  * @internal
  */
 export abstract class BaseCPProxy {
 
-    protected client: HazelcastClient;
     protected readonly proxyName: string;
     protected readonly serviceName: string;
     protected readonly groupId: RaftGroupId;
     protected readonly objectName: string;
+    protected readonly invocationService: InvocationService;
+    protected readonly serializationService: SerializationService;
 
-    constructor(client: HazelcastClient,
-                serviceName: string,
-                groupId: RaftGroupId,
-                proxyName: string,
-                objectName: string) {
-        this.client = client;
+    constructor(
```

Comment:
my ide says that we can use protected constructor for this class and for `CPSessionAwareProxy`

Reply:
Sure, why not.